src/main/java/seedu/mark/logic/commands/AutotagCommand.java

```diff
@@ -0,0 +1,52 @@
+package seedu.mark.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.mark.commons.util.CollectionUtil.requireAllNonNull;
+
+import seedu.mark.logic.commands.results.CommandResult;
+import seedu.mark.model.Model;
+import seedu.mark.model.autotag.SelectiveBookmarkTagger;
+import seedu.mark.storage.Storage;
+
+/**
+ * Adds an automatic tagger to Mark that tags bookmarks according to specific
+ * conditions.
+ */
+public class AutotagCommand extends Command {
+
+    public static final String COMMAND_WORD = "autotag";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Creates an automatic tagger that tags bookmarks according to the given conditions. At least one "
+            + "condition must be specified.\n"
+            + "Parameters: TAG_NAME [n/NAME] [u/PARTIAL_URL]\n"
```

Comment:
Just to clarify, what do you mean by match by name. Name of bookmark will be unique right? There will only be  at most one bookmark with the specified name then? Do you mean partial_name?

Reply:
Yep it's a partial match. I'll add a few lines in the UG to clarify what `NAME` and `PARTIAL_URL` are, and mention it here too.

I would prefer not to use `PARTIAL_NAME` as its meaning seems rather obscure (at least to me). On the other hand, `URL` would seem to imply the whole URL or at least domain, which is why `PARTIAL_URL` was chosen instead. If I explain that the autotag parameters `n/` and `u/` (etc.) are keywords which the name and URL should *contain*, do you think that `u/URL` would be sufficiently clear?

Reply:
`PARTIAL_URL` is fine. I was thinking about sth like `NAME_KEYWORD`, `URL_KEYWORD` so it is similar to the KEYWORD parameter in FIND command. Anw, as long as we explain it clearly in the user guide, it should be fine.

Reply:
Changed!